### PR TITLE
allow tests to run in two modes

### PR DESCRIPTION
**Pull request by devoncarew**
- allow tests to run in two modes (with either .packages or .dart_tool/package_config.json)

### CODE REVIEW

File: test/message_extraction/message_extraction_test.dart [hunk 1]
```diff
@@ -103,22 +103,33 @@ void copyFilesToTempDirectory() {
...
+    var sourcePackageConfig =
+        File(path.join('.dart_tool', 'package_config.json'));
+    var destPackageConfig =
+        File(path.join(tempDir, '.dart_tool', 'package_config.json'));
+    if (!destPackageConfig.parent.existsSync()) {
+      destPackageConfig.parent.createSync();
+    }
+    sourcePackageConfig.copySync(destPackageConfig.path);
+  } else {
+    // For google3, copy the .packages file.
```
Comment (natebosch) [inline test/message_extraction/message_extraction_test.dart:126]: Does it work to copy whichever file the current isolate is using for package config? https://api.dart.dev/stable/2.17.3/dart-isolate/Isolate/packageConfig.html

Comment (devoncarew) [reply test/message_extraction/message_extraction_test.dart:126]: I'll check

File: test/message_extraction/message_extraction_test.dart [hunk 2]
```diff
@@ -103,22 +105,28 @@ void copyFilesToTempDirectory() {
...
+  // Here we copy the package config file so the test can locate packages.
+  // For github, we use package_config.json; for google3, .packages.
+  final dartTool = '.dart_tool';
+
+  var configFile = File.fromUri(await Isolate.packageConfig);
+  var destFile = File(path.joinAll([
+    tempDir,
+    if (configFile.path.contains(dartTool)) dartTool,
+    path.basename(configFile.path),
+  ]));
```
Comment (natebosch) [inline test/message_extraction/message_extraction_test.dart:125]:
```suggestion
  var configFile = File.fromUri(await Isolate.packageConfig);
  var destFile = File(path.joinAll([
    tempDir,
    '.dartTool',
    'package_config.json',
  ]));
```

As far as I know we don't need to keep an identical layout. The file will contain json either way.

Comment (devoncarew) [reply test/message_extraction/message_extraction_test.dart]: That works (in both contexts)! Updated to always copy into package_config.json (w/ slightly different code that above).

**Review by natebosch — APPROVED** [review]

**Pull request merged**